test/services/ProveGateway/reactTo.test.ts

```diff
@@ -0,0 +1,174 @@
+      'proveGateway',
+      sinon.fake.resolves(fakeReceipt),
+    );
+    const proveGatewayService = new ProveGatewayService(
+      gateawayRepository as any,
+      messageRepository as any,
+      originWeb3,
+      auxiliaryWeb3,
+      auxiliaryWorkerAddress,
+      gatewayAddress,
+    );
+
+    const response = await proveGatewayService.reactTo(blockNumber);
+
+    SpyAssert.assert(
+      gateawayRepository.get,
+      1,
+      [[gatewayAddress]],
+    );
+
+    SpyAssert.assert(
+      messageRepository.isPendingMessages,
+      1,
+      [[blockNumber, gatewayAddress]],
+    );
+
+    SpyAssert.assert(
+      proofGeneratorStub,
+      1,
+      [[gatewayRecord.gatewayAddress, [], blockNumber]],
+    );
+
+    SpyAssert.assert(
+      coGatewayStub,
+      1,
+      [[
+        blockNumber,
+        proof.encodedAccountValue,
+        proof.serializedAccountProof,
+        { from: auxiliaryWorkerAddress }]],
+    );
+
+    assert.strictEqual(
+      fakeReceipt,
+      response.receipt,
+      'Service must return expected receipt',
+    );
+
+    assert.strictEqual(
+      response.success,
+      true,
+      'Success must be true',
+    );
+    assert.strictEqual(
+      response.message,
+      'Gateway successfully proven',
+      'Service must return correct messages',
+    );
+  });
+
+  it('should fail to react if gayeway details does not exists', async (): Promise<void> => {
+    const gateawayRepository = sinon.createStubInstance(GatewayRepository, {
+      get: Promise.resolve(null),
+    });
+
+    const proveGatewayService = new ProveGatewayService(
+      gateawayRepository as any,
+      sinon.fake() as any,
+      originWeb3,
+      auxiliaryWeb3,
+      auxiliaryWorkerAddress,
+      gatewayAddress,
+    );
+
+    await assert.isRejected(
+      proveGatewayService.reactTo(
+        blockNumber,
+      ),
+      'Gateway record record doesnot exists for given gateway',
+      'It must fail if gatway record does not exists.',
+    );
+  });
+
+  it('should not try to proveGateway if there are no pending messages', async (): Promise<void> => {
+    const gatewayRecord: Gateway = StubData.gatewayRecord();
+    const gateawayRepository = sinon.createStubInstance(GatewayRepository, {
+      get: Promise.resolve(gatewayRecord),
+    });
+
+    const messageRepository = sinon.createStubInstance(MessageRepository, {
+      isPendingMessages: Promise.resolve(false),
+    });
+
+    const proveGatewayService = new ProveGatewayService(
+      gateawayRepository as any,
+      messageRepository as any,
+      originWeb3,
+      auxiliaryWeb3,
+      auxiliaryWorkerAddress,
+      gatewayAddress,
+    );
+
+    const response = await proveGatewayService.reactTo(blockNumber);
+
+    SpyAssert.assert(
+      gateawayRepository.get,
+      1,
+      [[gatewayAddress]],
+    );
+
+    SpyAssert.assert(
+      messageRepository.isPendingMessages,
+      1,
+      [[blockNumber, gatewayAddress]],
+    );
+
```

Comment:
Should we be also asserting that `getOutboxProof` and `proveGateway` weren't called ?